### DISCUSSION TITLE
Column support notes?

### DISCUSSION THREAD

**Discussion by Tordai079** [TriliumNext/Trilium#4654] [General]
I was wondering it supports columns in the text note. Simply trying to make a Cornell note format.

**Comment by SiriusXT** [comment]
Trilium does not support multiple columns. One way to compensate is to use multiple columns in the table.

**Comment by rom1dep** [comment]
This would probably require migrating to a new editor, which could be on scope for Trilium Next: https://github.com/orgs/TriliumNext/discussions/24 (FTR, lexical.dev seems to support this feature)

**Comment by pancake-tongs** [comment]
Coming from OneNote (and wanting to get away from it desperately), the lack of support for multiple columns and arbitrary text placement are a major problem. I've got 10 years of notes, and nearly all of them make use of these features. For that reason, I can't migrate to Trilium even though I want to.

**Reply by eliandoran** [reply]
Unfortunately there's nothing we can do about it for now. 

OneNote is very different in philosophy compared to Trilium, where OneNote has a freeflow format (which some people prefer or not). Personally I've never found much use for the layout system of OneNote, it felt cumbersome. 

Could you offer some screenshots of how your notes are structured in columns?

**Reply by rom1dep** [reply]
Have you tried to use Canvas notes instead? As the name implies, they let you place your content (text, images, …) on an infinite canvas.

**Reply by pancake-tongs** [reply]
> Have you tried to use Canvas notes instead? As the name implies, they let you place your content (text, images, …) on an infinite canvas.

I had - and it has promise, but IIRC in that mode most of the options for advanced text formatting are absent, so it solves the placement issue but introduces other issues. It seems clear that while canvas overlaps a bit with what I'm asking for, that is not really what it was designed to do.


> OneNote is very different in philosophy compared to Trilium, where OneNote has a freeflow format (which some people prefer or not). Personally I've never found much use for the layout system of OneNote, it felt cumbersome.
> 
> Could you offer some screenshots of how your notes are structured in columns?

I understand the differing philosophy, though Trillium's support for plugins gives me hope that use cases like mine can be accommodated in anything other than OneNote. A ton of people write markdown every day, and no doubt have no complaints about using the same for their personal notes. I got started on Onenote a long time ago, and simply adapted my workflow to that application, for better or worse. I imagine there must be many others like me, but despite looking rather hard, I've never found an alternative with equivalent functionality. It's a gap that trillium might be able to fill.

I'm happy to share a screenshot to illuminate the reason for this ask, but I'm not looking for critique of my note-taking system.

I use OneNote to organize my work week and keep track of things, and for more detailed notes as needed. The screenshot below is heavily redacted but demonstrates my use. I create a free-floating block for each day of the week, load it with my meetings and goals for the day. I also prepare my work hours to be logged and might add technical details of a given issue to one of these pages. If I need to take detailed notes relevant to a workday item, I'll create a sub-page for them. I also keep notebooks with more general information. In this way I end up with a nice linear sequence of notes from my job as well as more detailed pages as needed.

<img width="3163" height="1639" alt="Onenote_censored" src="https://github.com/user-attachments/assets/b245dfa0-574f-42e6-b7f8-4f4f879dc0d4" />

**Reply by eliandoran** [reply]
> I'm happy to share a screenshot to illuminate the reason for this ask, but I'm not looking for critique of my note-taking system.

It was not my intention to critique the note-taking system, it's simply that Trilium is not built with this philosophy in mind which is quite specific to OneNote.

Canvas indeed doesn't seem suitable for day-to-day use.

One thing you could try is to use a table:

* Create a 3x2 table.
* Adjust the border of the table to have a `transparent` color.
* Merge columns where necessary.
* Use nested tables for the Test Cases and Hours section.

You can then create a template out of it. Afterwards you can also look into week notes, which should help you to generate the weekly notes (although day notes might get in the way).

**Reply by pancake-tongs** [reply]
I appreciate your responsiveness. I'll give this a shot soon and report back. I'm eager enough to get away from oneNote to endure a few workarounds. 


> It was not my intention to critique the note-taking system, 

Sorry, I guess I was feeling defensive. You have not been critical.

**Comment by rodrigoalexandrecosta** [comment]
The CKEditor has a layout table feature (it's not premium): https://ckeditor.com/docs/ckeditor5/latest/features/tables/layout-tables.html

Question: would it be possible to bring this functionality to Trilium?

**Reply by eliandoran** [reply]
@rodrigoalexandrecosta , I know about it but we didn't integrate it yet since we don't feel like it can bring much of a benefit.
What would be a feature of the layout table that can't be replicated using a normal one?

**Reply by rodrigoalexandrecosta** [reply]
Sorry for the late answer. I guess that the specific feature of layout table feature attracted me because I'm not able to create a table without borders. Changing the borders to "invisible" doesn't work for me, thus I can create a table with two or more columns but that doesn't seamless integrate  with a proper text note. I'll try to understand why this is the case and, if it's the case, open an issue.

**Reply by eliandoran** [reply]
@rodrigoalexandrecosta , invisible borders are supported. Please have a look at https://docs.triliumnotes.org/User%20Guide/User%20Guide/Note%20Types/Text/Tables.html#tables-with-invisible-borders

**Reply by rodrigoalexandrecosta** [reply]
Yeah, but it doesn't seem to work for me. Both table and cell are with borders=none, for cells it works, but not for the outer borders of the table. Don't know if it's a bug with the CKEditor, but it's always been like that (I've been using Trilium for years)

<img width="999" height="474" alt="image" src="https://github.com/user-attachments/assets/a94d7379-d02e-4362-b86b-f260b471dbd6" />

**Reply by eliandoran** [reply]
@rodrigoalexandrecosta , in the link I've provided it mentions that you need to set the borders to single for this to work. 

> In Table properties, set:
>     Style to Single
>     Color to transparent
>     Width to 1px.

And then repeat for cell properties.
